hierarchy/hierarchy_test.go

```diff
@@ -185,90 +185,138 @@ func getMockHierarchyAPI(expectRequest http.Request, mockedHTTPResponse MockedHT
 func TestClient_GetRoot(t *testing.T) {
 	instanceID := "foo"
 	name := "bar"
-	model := `{"label":"","has_data":true}`
-	Convey("When bad request is returned", t, func() {
+
+	Convey("Given a bad request API response, then the expected error is returned", t, func() {
 		ts, mockedAPI := getMockHierarchyAPI(http.Request{Method: "GET"}, MockedHTTPResponse{StatusCode: 400, Body: ""})
 		_, err := mockedAPI.GetRoot(ctx, instanceID, name)
-		So(err, ShouldNotBeNil)
+		So(err, ShouldResemble, &ErrInvalidHierarchyAPIResponse{
+			actualCode:   400,
```

Comment:
Perhaps throughout the tests we could use the Go consts for the http codes? E.g. `http.StatusBadRequest` here?

Reply:
Yes, good point. I can do the same for all the status codes